akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonActions.scala

```diff
@@ -32,6 +32,9 @@ private[parser] trait CommonActions {
         case "encrypted"   ⇒ multipart.encrypted(params)
         case custom        ⇒ MediaType.customMultipart(custom, params)
       }
+      case "*" if subType != "*" ⇒
+        throw new IllegalArgumentException(s"Unsupported */$subType MediaType detected, " +
```

Comment:
I like the error message but we should fix it directly in the model, otherwise we might have missed other problematic code paths.